client/hwlib/src/collectors/os_info.rs

```diff
@@ -18,21 +18,35 @@
  *        Nadzeya Hutsko <nadzeya.hutsko@canonical.com>
  */
 
-use anyhow::{anyhow, Context, Result};
+use anyhow::{anyhow, Result};
 use lazy_static::lazy_static;
 use regex::Regex;
 use std::{fs::read_to_string, path::Path, process::Command};
 
 use crate::models::software::{KernelPackage, OS};
 
-impl TryFrom<&Path> for OS {
+pub trait CommandRunner {
```

Comment:
This is an interesting question. The `CertificationStatusRequest` struct is public, because it's used to prepare the request and collect the hardware data. It has the `pub` method `new`, which accepts only `&Paths` to the files that we can use to collect the data
Since OS module needs `CommandRunner` trait to execute commands, as @sminez indicated suggested, I added the `new_with_runner` method, which accepts both `&Paths` and `&impl CommandRunner`. The `new` method calls the `new_with_runner` method, using the `RealCommandRunner`.

And here is the dilemma:
- We make the `new_with_runner` method private or `pub(crate)`, which seems to make sense. However, as I understand (but not sure), it'll make mocking the `CommandRunner` in the integration tests impossible, since we won't be able to call the `pub(crate)` method from the `tests/` folder outside the source code (but I may be wrong)
- We make the `new_with_runner` method `pub`, but it'll require making the `CommandRunner` `pub` as well.

Reply:
Ahhh I see, thanks for the detailed explanation! The only nice thing I can think of would be to move the integration tests into a `mod tests` in `lib.rs` as it's impossible to test without this internal detail. This is unfortunate, I agree that exposing this is awkward, but I don't think it's _too_ bad either way. For pragmatism, the existing visibility seems okay to me

Reply:
I guess it's better to place the tests in `lib.rs` instead of exposing this trait. If we have issues with tests in future, we can make it `pub`